src/v3/src/hooks/usePolling.test.tsx

```diff
@@ -109,6 +109,112 @@ describe('usePolling', () => {
+      // expect(mockProceedFn).toHaveBeenCalled();
+      expect(mockProceedFn).toHaveBeenCalledWith({
+        stateHandle: undefined,
+        autoChallenge: false,
+        step: 'challenge-poll',
+      });
+
+      // expect to return new polling transaction
+      await waitFor(() => {
+        expect(result.current).toMatchObject({
+          nextStep: {
+            name: 'challenge-poll',
+            refresh: 4000,
+          },
+        });
+      });
+    });
+
+    it('should trigger poll request with autoChallenge=true when data contains user selected value', async () => {
+      // @ts-ignore remove after adding refresh to nextStep in auth-js
+      const idxTransaction = {
+        nextStep: {
+          name: 'challenge-poll',
+          refresh: 4000,
+        },
+      } as IdxTransaction;
+      mockData.autoChallenge = true;
+      const { result } = renderHook(() => usePolling(idxTransaction, mockProps, mockData));
+
+      // expect to setup timer
+      expect(setTimeout).toHaveBeenCalledTimes(1);
+      expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 4000);
+
+      // expect to call action function when timeout
+      jest.advanceTimersByTime(4000);
+      // expect(mockProceedFn).toHaveBeenCalled();
+      expect(mockProceedFn).toHaveBeenCalledWith({
+        stateHandle: undefined,
+        autoChallenge: true,
+        step: 'challenge-poll',
+      });
+
+      // expect to return new polling transaction
+      await waitFor(() => {
+        expect(result.current).toMatchObject({
+          nextStep: {
+            name: 'challenge-poll',
+            refresh: 4000,
+          },
+        });
+      });
+    });
+
+    it('should trigger poll request with autoChallenge=true when data contains user selected value and transaction contains different value', async () => {
+      // @ts-ignore remove after adding refresh to nextStep in auth-js
+      const idxTransaction = {
+        nextStep: {
+          name: 'challenge-poll',
+          refresh: 4000,
+          inputs: [{ name: 'autoChallenge', value: false }],
+        },
+      } as IdxTransaction;
+      mockData.autoChallenge = true;
+      const { result } = renderHook(() => usePolling(idxTransaction, mockProps, mockData));
+
+      // expect to setup timer
+      expect(setTimeout).toHaveBeenCalledTimes(1);
+      expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 4000);
+
+      // expect to call action function when timeout
+      jest.advanceTimersByTime(4000);
+      // expect(mockProceedFn).toHaveBeenCalled();
+      expect(mockProceedFn).toHaveBeenCalledWith({
+        stateHandle: undefined,
+        autoChallenge: true,
+        step: 'challenge-poll',
+      });
+
+      // expect to return new polling transaction
+      await waitFor(() => {
+        expect(result.current).toMatchObject({
+          nextStep: {
+            name: 'challenge-poll',
+            refresh: 4000,
+          },
+        });
+      });
+    });
+
```

Comment:
nit: `yarn jest --coverage usePolling.test.tsx` shows that

```ts
```

Reply:
After attempting to write a test for this, I realized this code is unreachable. There is no way to set the timerRef value when initializing the test, plus, clearTimeout accepts an id or undefined, so even if timerRef.current is undefined there is no need to explicitly check it before calling clearTimeout. I am removing the if check branch.